.github/workflows/package.yaml

```diff
@@ -91,6 +91,34 @@ jobs:
         name: Rancher Desktop-linux.zip
         path: dist/rancher-desktop*.zip
         if-no-files-found: error
+    - name: Copy linux zip to canary for main
+      uses: canastro/copy-file-action@0.0.2
+      if: ${{ startsWith(matrix.os, 'ubuntu-') && github.ref_name == 'main' }}
+      with:
+        source: "dist/rancher-desktop*.zip"
+        target: "dist/rancher-desktop-canary.zip"
+    - name: Copy canary to s3
+      uses: prewk/s3-cp-action@v2
+      if: ${{ startsWith(matrix.os, 'ubuntu-') && github.ref_name == 'main' }}
+      with:
+        aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        source: './dist/rancher-desktop-canary.zip'
+        dest: 's3://rancher-desktop-assets-for-obs/rancher-desktop-canary.zip'
+    - name: Copy linux zip to release for tags
+      uses: canastro/copy-file-action@0.0.2
+      if: ${{ startsWith(matrix.os, 'ubuntu-') && github.ref_type == 'tag' }}
```

Comment:
Are we sure we want that for all tags? AFAIK we tag, stuff gets build and then there is a manual release process. Wouldn't be better to trigger this upload for an actual release? I would have added a new release workflow that gets triggered after then manual release process is done, so we _ensure_ everything is in sync (there can't be an automated linux release without a release at github). So handle its own workflow with something like.

```yaml
on:
  release:
    types: [prereleased, released]

jobs:
  linux-release:
    runs-on: ubuntu-latest
    steps:
    ## 1. Download published release based on tag ($GITHUB_REF), this is a predictable URL
    ## 2. Upload the release to s3://rancher-desktop-assets-for-obs/rancher-desktop-release.zip
```

Reply:
Good idea. I've updated. Using the published event as it will pick up prereleases that come out of drafts. The prerelease event doesn't do that, according to the docs.

Reply:
Great, I did not dive into details of what triggers what :+1: